osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs

```diff
@@ -302,7 +302,7 @@ private void endUserDrag()
         /// <summary>
         /// The total amount of time visible on the timeline.
         /// </summary>
-        public double VisibleRange => track.Length / Zoom;
+        public double VisibleRange => (DisplayableContent / Content.DrawWidth) * track.Length;
```

Comment:
I was very confused as to why this was apparently "more correct" than the old version. For all intents and purposes, `DisplayableContent / Content.DrawWidth` should be precisely equal to `1 / Zoom`:

https://github.com/ppy/osu/blob/83c2b27b693e2ccf3ba5e1403c8736cd21d196c2/osu.Game/Screens/Edit/Compose/Components/Timeline/ZoomableScrollContainer.cs#L131

I believe that the reason why `1 / Zoom` is not working correctly, is that the bug is actually in `ZoomableScrollContainer` - the container is not responding to draw size invalidations, and therefore changing the window size is potentially inadvertently changing the zoom level, therefore breaking the `VisibleRange` calculation too. Here's a video - captured on this branch - showing why I think this:

https://user-images.githubusercontent.com/20418176/170878692-d03fb2db-3b0e-4e3c-be11-be3d15740199.mp4

Note how the zoom level snaps abruptly when I just nudge the buttons ever so slightly. The working theory is that when the window is being resized, the value of `DisplayableContent` changes, but `Content.DrawWidth` stays the same, which leads to `Zoom` having a stale/incorrect value.

Reply:
Ah, good find. Should probably have it fixed in this PR instead, will take a look whenever I get the chance.

Reply:
Have reverted the current change and fixed `ZoomableScrollContainer` to update its width on external size changes.